src/litserve/loops.py

```diff
@@ -300,7 +323,11 @@ def run_batched_streaming_loop(
     max_batch_size: int,
     batch_timeout: float,
     callback_runner: CallbackRunner,
+    concurrent_decode: bool = True,
 ):
+    if concurrent_decode:
+        executor = concurrent.futures.ThreadPoolExecutor(max_workers=os.cpu_count())
```

Comment:
We can limit number of threads to batch size
```suggestion
        executor = concurrent.futures.ThreadPoolExecutor(max_workers=min(max_batch_size, os.cpu_count()))
```